jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -195,8 +198,10 @@ def search(self, error_on_fail=False):
             # but this exceptions might be raised here if setting from_file from the UI
             table = self.catalog.selected_obj
             self.app._catalog_source_table = table
-            skycoord_table = table['sky_centroid']
-
+            if len(table['sky_centroid']) > self.max_sources:
+                skycoord_table = table['sky_centroid'][:self.max_sources]
```

Comment:
same as above - either snackbar or in-plugin warning would be nice (or as a follow-up)